client/state/data-layer/wpcom/sites/memberships/index.js

```diff
@@ -65,14 +65,21 @@ export const handleMembershipProductsList = dispatchRequest( {
 			action
 		),
 	fromApi: function ( endpointResponse ) {
+		// If the response has an error, return an empty array for products.
+		if ( endpointResponse.hasOwnProperty( 'error' ) ) {
+			return [];
+		}
+
 		const products = endpointResponse.products.map( membershipProductFromApi );
 		return products;
 	},
-	onSuccess: ( { siteId }, products ) => ( {
-		type: MEMBERSHIPS_PRODUCTS_RECEIVE,
-		siteId,
-		products,
-	} ),
+	onSuccess: ( { siteId }, products ) => {
+		return {
+			type: MEMBERSHIPS_PRODUCTS_RECEIVE,
+			siteId,
+			products,
+		};
+	},
```

Comment:
```suggestion
	onSuccess: ( { siteId }, products ) => ( {
		type: MEMBERSHIPS_PRODUCTS_RECEIVE,
		siteId,
		products,
	} ),
```

Reply:
Changes seem not necessary here, so reverting...

Reply:
The changes were made for debugging but not restored. 😅